src/main/scala/com/virtuslab/shared_indexes/Main.scala

```diff
@@ -146,26 +158,87 @@ object Main {
     }
   }
 
-  private def generateJarSharedIndexesFromProjectDeps(
-    intellij: IntelliJ,
-    workspace: Workspace,
-    projectRoot: os.Path
-  ): Unit = {
-    if (os.list(workspace.jarIndexes).isEmpty) {
-      println(s"Extracting dependency JARs from project ${projectRoot.baseName}")
-      generateJarSharedIndexes(intellij, workspace, JarLocator.getSbtDeps(projectRoot))
-    } else {
-      println("JARs indexes already exist, skipping")
+  private def findJarsToIndex(workPlan: WorkPlan): Seq[os.Path] = {
+    import workPlan._
+    var jarPaths = if (projectRoot.isDefined) JarLocator.getSbtDeps(projectRoot.get) else artifactPaths
```

Comment:
Pattern matching to get each of the three cases would be more readable IMO

Reply:
Done